lib/collection/property-list.js

```diff
@@ -289,6 +316,40 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
     },
 
     /**
+     * Add or update values from a source list.
+     *
+     * @param {PropertyList|Array} source
+     * @param {Boolean=false} prune Setting this to `true` will cause the extra items from the list to be deleted
+     */
+    assimilate: function (source, prune) {
+        if (PropertyList.isPropertyList(source)) {
+            source = source.members;
```

Comment:
It would be nice to assign source.members to a fresh variable instead of changing the type on the fly.

Reply:
Can you elaborate @kunagpal ?

Reply:
Within the `if` block, `source` is an instance of `PropertyList`, which then gets changed to an `Array`. This will cause an optimization bailout within the `assimilate` function.